components/SQLEditor.go

```diff
@@ -77,3 +82,95 @@ func (s *SQLEditor) SetBlur() {
 	s.SetBorderColor(tcell.ColorWhite)
 	s.SetTextStyle(tcell.StyleDefault.Foreground(tcell.ColorWhite))
 }
+
+
+/*
+	THIS FUNCTION OPENS EXTERNAL EDITOR.
+
+	Q: WHY OPEN ANOTHER TERMINAL?
+	A: OPENING EDITORS LIKE VIM/NEOVIM REALLY MESSED UP INITIAL TERMINAL'S OUTPUT.
+*/
+
+func openExternalEditor(s *SQLEditor) string {
+	// Path of temporary file
+	path := "/tmp/lazysql.sql"
+
+	editor := getEditor()
+	terminal := getTerminal()
+
+	// Create a temporary file with the current SQL query content
+	content := []byte(s.TextArea.GetText())
+
+	/*
+	0644 Permission
+	* User: read & write
+	* Group: read
+	* Other: read
+	*/
+
+	err := os.WriteFile(path, content, 0644)
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Setup command
+	cmd := exec.Command(terminal, "-e", editor, path)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+
+	// cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Read the updated content from the temporary file
+	updatedContent, err := os.ReadFile(path)
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Remove the temporary file
+	err = os.Remove(path)
+
+	if err != nil {
+		// TODO: Handle error
+	}
+
+	// Convert to string before returning
+	return string(updatedContent)
+}
+
+// Function to select editor
+func getEditor() string {
+	var editor string = os.Getenv("SQL_EDITOR")
+	if editor == "" {
+		editor = os.Getenv("EDITOR")
+	}
+
+	if editor == "" {
+		editor = os.Getenv("VISUAL")
+	}
+
+	if editor == "" {
+		editor = "vi" 		// use "vi" if $EDITOR not set
+	}
+
+	return editor
+}
+
+// Function to select terminal
+func getTerminal() string {
+	var terminal string = os.Getenv("SQL_TERMINAL")
+
+	if terminal == "" {
+		terminal = os.Getenv("TERMINAL")
```

Comment:
I have never seen this variable I think

Reply:
Well I've tested it on my machine and it does exist.